clippy_lints/src/methods/mod.rs

```diff
@@ -2185,6 +2193,22 @@ fn lint_map_or_none<'a, 'tcx>(cx: &LateContext<'a, 'tcx>, expr: &'tcx hir::Expr,
     }
 }
 
+fn lint_filetype_is_file(cx: &LateContext<'_, '_>, expr: &hir::Expr, args: &[hir::Expr]) {
+    const LINT_MSG: &str = "`FileType.is_file()` does not cover all readable file types in linux.";
+    const HELP_MSG: &str = "Use `!FileType.is_dir()` instead.";
```

Comment:
```suggestion
    const HELP_MSG: &str = "use `!FileType.is_dir()` instead";
```

This could maybe be an auto applicable suggestion. This mustn't be done in this PR though.